Sources/BraintreeVisaCheckout/BTVisaCheckoutClient.swift

```diff
@@ -1,6 +1,121 @@
+                /// TODO: Add failure analytics event
+                completion(nil, error)
+                return
+            }
+
+            guard let configuration, let json = configuration.json else {
+                /// TODO: Add `fetchConfigurationFailed` analytics event
+                completion(nil, error)
+                return
+            }
+
+            guard json["visaCheckout"]["apiKey"].isTrue else {
+                /// TODO: Add failure analytics Visa Checkout not enabled
+                return
+            }
+
+            let environmentString = configuration.visaCheckoutEnvironment
+            let environment: Environment = environmentString == "sandbox" ? .sandbox : .production
+
+            let profile = Profile(
+                environment: environment,
+                apiKey: configuration.visaCheckoutAPIKey ?? "",
+                profileName: nil
+            )
+            profile.datalevel = .full
+            profile.clientId = configuration.visaCheckoutExternalClientID
+            profile.acceptedCardBrands = configuration.acceptedCardBrands
+
+            completion(profile, nil)
+        }
+    }
+
+    /// Tokenizes a Visa checkout result.
+    /// - Note: The `checkoutResult` parameter is declared as `callID` type, but you must pass a `VisaCheckoutResult` instance.
+    /// - Parameters:
+    ///   - checkoutResult: A Visa `CheckoutResult` instance.
+    ///   - completion: A completion block that is invoked when tokenization has completed. If tokenization succeeds,
+    ///   `tokenizedVisaCheckoutCard` will contain a nonce and `error` will be `nil`; if it fails
+    ///   `tokenizedVisaCheckoutCard` will be `nil` and `error` will describe the failure.
+    @objc public func visaPaymentSummary(
+        _ checkoutResult: CheckoutResult,
+        completion: @escaping (BTVisaCheckoutNonce?, Error?) -> Void
+    ) {
+        let statusCode = checkoutResult.statusCode
+        let callID = checkoutResult.callId
+        let encryptedKey = checkoutResult.encryptedKey
+        let encryptedPaymentData = checkoutResult.encryptedPaymentData
+
+        tokenize(
+            statusCode: statusCode,
+            callID: callID,
+            encryptedKey: encryptedKey,
+            encryptedPaymentData: encryptedPaymentData,
+            completion: completion
+        )
+    }
+
+    /// Tokenizes a Visa checkout result. Exposed for testing properties of the Visa `CheckoutResult`
+    /// - Parameters:
+    ///   - statusCode: The result code indicating the status of a completed Visa Checkout transaction.
+    ///   - callId: The unique identifier for the Visa Checkout transaction.
+    ///   - encryptedKey: The encrypted key associated with the Visa Checkout transaction.
+    ///   - encryptedPaymentData: The encrypted payment data for the Visa Checkout transaction.
+    ///   - completion: A completion block that is invoked when tokenization has completed.
+    @objc public func tokenize(
```

Comment:
I don't think we need a separate method for this. Can we just move this logic above?